core/parentchain/light-client/src/io.rs

```diff
@@ -80,6 +80,11 @@ where
 
 	if genesis_hash == params.genesis_header.hash() {
 		validator.set_state(validation_state);
+		// The init_parachain_validator function clear the state every time,
+		// so we should write the state again.
+		LightClientStateSeal::<B, LightValidationState<B>>::seal_to_static_file(
+			validator.get_state(),
+		)?;
```

Comment:
I see thanks for the solution, but according to your description, this seems like a hack. Is this correct? If yes, I would do a follow-up issue to properly fix this.

Reply:
Hi @clangenb , I think that's the right way to go - it has to be persisted on disk to survive reboot. If you have better ideas, please let me know

Reply:
Yes, but in my opinion, the problem here is the the `init_parachain_validator` overwrites the previously sealed light client, and the entire initialization flow might need to be revised. Anyhow, I think that the light-client will be soon refactored anyhow, so I think that this is totally ok for now.

Reply:
I see, thanks. It's great to hear that there's already a plan to revise and refactor the flow :)